### PR TITLE
Implement Bar Chart type 

### PR DESCRIPTION
**Pull request by chaliya96**
### Type of change

Please tick options that are relevant.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

## Changes
- Implement Bar Chart type in feature viewer charts. 

## Flags
1. Requires some refactoring.

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/feature-viewer.js [hunk 1]
```diff
@@ -1165,7 +1223,54 @@ function createFeature(sequence, div, options) {
...
-                
+                forcePropagation(histog);
+            },
+            bar: function (object, position) {
+                if (object.fill === undefined) object.fill = true;
+                var histog = svgContainer.append("g")
+                    .attr("class", "bar")
+                    .attr("transform", "translate(0," + position + ")");
+                var dataline=[];
+                dataline.push([{
```
Comment (kwsamarasinghe) [inline src/feature-viewer.js:1234]: Why is this definition of dataline[]? Shouldn't it be passed in the feature deifinition?

File: src/feature-viewer.js [hunk 2]
```diff
@@ -1165,7 +1223,54 @@ function createFeature(sequence, div, options) {
...
+                    .attr("clip-path", "url(#clip)")
+                    .attr("d", lineGen)
+                    .attr("class", "line" + object.className)
+                    .style("z-index", "0")
+                    .style("stroke", "black")
+                    .style("stroke-width", "1px");
+                    object.data.forEach(function(dd,i,array){
+                    histog.selectAll()
+                    .data(dd)
+                  .enter().append("rect")
```
Comment (kwsamarasinghe) [inline src/feature-viewer.js]: Check indentation

File: src/feature-viewer.js [hunk 3]
```diff
@@ -1392,6 +1497,29 @@ function createFeature(sequence, div, options) {
...
+                }
+                transit
+//                    .data(object.data)
+                    //.transition()
+                    //.duration(500)
+                    .attr("x", function (d) {
+                        return scaling(d.x - 0.4)
+                    })
+                    .attr("width", function (d) {
+                        if (scaling(d.x + 0.4) - scaling(d.x - 0.4) < 2) return 2;
```
Comment (kwsamarasinghe) [inline src/feature-viewer.js:1519]: How does this 0.4 determined?

File: src/feature-viewer.js [hunk 4]
```diff
@@ -690,6 +730,24 @@ function createFeature(sequence, div, options) {
...
+                    object.data.forEach(function(d){
+                        if (d.filter(function(l){ return l.y < 0}).length) negativeNumbers = true;
+                    });
+                    preComputing.bar(object);
+                    
+                    fillSVG.bar(object, Yposition);
+                    Yposition += pathLevel;
+                    yData.push({
+                        title: object.name,
+                        y: Yposition - 10,
```
Comment (kwsamarasinghe) [inline src/feature-viewer.js:746]: How does this 10 is chosen?

File: src/feature-viewer.js [hunk 5]
```diff
@@ -622,6 +631,37 @@ function createFeature(sequence, div, options) {
...
+                        })
+                    }
+                    var maxValue = Math.max.apply(Math,object.data[i].map(function(o){return Math.abs(o.y);}));
+                    level = maxValue > level ? maxValue : level;
+                }
+                lineYscale.range([0, -(shift)]).domain([0, -(level)]);
+                pathLevel = shift * 10 +5;
+                object.maxValue = maxValue;
+                object.level = level;
+                object.shift = shift * 10 +5;
```
Comment (kwsamarasinghe) [inline src/feature-viewer.js:662]: How is "shift * 10 + 5" is determined

**Review by kwsamarasinghe — CHANGES_REQUESTED** [review]
Please check my comments, specially on the constants used in the equations e.g "shift * 10 + 5". Make sure these constants cover all (at least most ) of the bar charts.  Also please remove commented lines in the new code.

File: src/feature-viewer.js (hunk 1 above)
Comment (chaliya96) [reply src/feature-viewer.js:1234]: This is related to the X axis required data is passed in later. The implementation is similar to the line chart type X axis.

File: src/feature-viewer.js (hunk 2 above)
Comment (chaliya96) [reply src/feature-viewer.js]: Corrected.

File: src/feature-viewer.js (hunk 3 above)
Comment (chaliya96) [reply src/feature-viewer.js:1519]: Similar implementation used in Rect chart type..

File: src/feature-viewer.js (hunk 4 above)
Comment (chaliya96) [reply src/feature-viewer.js:746]: Similar implementation used in line chart type. 10 is the constant value which used in all the chart types.

File: src/feature-viewer.js (hunk 5 above)
Comment (chaliya96) [reply src/feature-viewer.js:662]: Similar implementation used in line chart type.  * 10  + 5 are the constant value which used in all the chart types.

**Review by kwsamarasinghe — APPROVED** [review]

**Pull request merged**